src/main/java/com/Service/EventService.java

```diff
@@ -0,0 +1,38 @@
+package com.Service;
+
+import com.Model.Event;
+import com.Model.Patient;
+import com.Model.PatientEventAssociation;
+import com.Repository.EventRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class EventService {
+
+    @Autowired
+    private EventRepository eventRepo;
+
+    public void saveEvents (List<Event> events) throws Exception{
+        Patient tempPatient = new Patient();//created patient
+        List<Event> eventList = new ArrayList<>();
+        PatientEventAssociation association;
+        for (Event e: events){
+            eventList = this.eventRepo.findAllByNameAndDate(e.getName(), e.getDate());
+            int index = eventList.indexOf(e);
+            if (index<0) {//new Event, save it to Repo
+                association = new PatientEventAssociation(e, tempPatient, true);
```

Comment:
should the association always be "true"? There is no test here as to whether the caller was a verified user.

Reply:
For now I still have not implemented VerifiedUser yet, I am changing the default association to "false". Once we have VerifiedUser implemented, I will add logic here.